pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -386,14 +429,17 @@ private static class PipelineThread extends Thread {
     public void run() {
       try {
         FhirEtlOptions options = pipeline.getOptions().as(FhirEtlOptions.class);
-        EtlUtils.runPipelineWithTimestamp(pipeline, options);
+        PipelineResult pipelineResult = EtlUtils.runPipelineWithTimestamp(pipeline, options);
+        manager.publishPipelineMetrics(pipelineResult.metrics());
         if (mergerOptions == null) { // Do not update DWH yet if this was an incremental run.
           manager.updateDwh(options.getOutputParquetPath());
         } else {
           FhirContext fhirContext = FhirContexts.forR4();
           Pipeline mergerPipeline = ParquetMerger.createMergerPipeline(mergerOptions, fhirContext);
           logger.info("Merger options are {}", mergerOptions);
-          EtlUtils.runMergerPipelineWithTimestamp(mergerPipeline, mergerOptions);
+          PipelineResult mergerPipelineResult =
+              EtlUtils.runMergerPipelineWithTimestamp(mergerPipeline, mergerOptions);
+          manager.publishPipelineMetrics(mergerPipelineResult.metrics());
```

Comment:
So in this case we will delete the metrics from the above incremental run, correct? I think for the merger pipeline we should not call `removePipelineMetrics()`.